src/js/utils/ui.js

```diff
@@ -0,0 +1,128 @@
+define([
+    'utils/backbone.events',
+    'events/events',
+    'utils/underscore'
+], function(Events, events, _) {
+    var ui = function (elem, options) {
+        var _elem = elem,
+            _enableDoubleTap = (options && options.enableDoubleTap),
+            _hasMoved = false,
+            _lastClickTime = 0,
+            _doubleClickDelay = 300;
+
+        elem.addEventListener('touchstart', interactStartHandler);
+        elem.addEventListener('mousedown', interactStartHandler);
+
+        function interactStartHandler() {
+            elem.addEventListener('touchmove', interactDragHandler);
+            elem.addEventListener('touchcancel', interactEndHandler);
+            elem.addEventListener('touchend', interactEndHandler);
+
+            document.addEventListener('mousemove', interactDragHandler);
+            document.addEventListener('mouseup', interactEndHandler);
+        }
+
+        function interactDragHandler(evt) {
+            var touchEvents = events.touchEvents;
+
+            if (_hasMoved) {
+                triggerEvent(touchEvents.DRAG, evt);
+            } else {
+                triggerEvent(touchEvents.DRAG_START, evt);
+                _hasMoved = true;
+                triggerEvent(touchEvents.DRAG, evt);
+            }
+        }
+
+        function interactEndHandler(evt) {
+            var touchEvents = events.touchEvents;
+
+            elem.removeEventListener('touchmove', interactDragHandler);
+            elem.removeEventListener('touchcancel', interactEndHandler);
+            elem.removeEventListener('touchend', interactEndHandler);
+
+            document.removeEventListener('mousemove', interactDragHandler);
+            document.removeEventListener('mouseup', interactEndHandler);
+
+            if (_hasMoved) {
+                triggerEvent(touchEvents.DRAG_END, evt);
+            } else {
+                // This allows the controlbar/dock/logo click events not to be forwarded to the view
+                evt.cancelBubble = true;
```

Comment:
Looks like we can't use this : https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/cancelBubble

why not do preventDefault()?